### PR TITLE
Fix:  'internal error: no storage type for block output'

### PR DESCRIPTION
**Pull request by reveriejake**
Fixes errors which appear when loading in state data through variables.

Works on my end. Please double check before accepting pull request.

Original solution provided by: https://github.com/Akeyn

### CODE REVIEW

**Comment by reveriejake** [comment]
I tried to play it safe with defines. Let me know if this is not necessary. Not sure how universal [flatten] and [unroll()] are.

**Comment by reveriejake** [comment]
I tried to use their defines but it doesn't seem to recognize them

UNITY_EDITOR || UNITY_STANDALONE

**Comment by reveriejake** [comment]
I've tried a few of the Unity defines but they didn't seem to get recognized. I didn't see any HLSL specific directives that would work either. I could be wrong, I will keep an eye out for that.

If these [flatten] and [unroll()] tags are Unity specific, it might make more sense to rename ENABLE_FLATTEN to UNITY_ENGINE or something.

**Comment by Auburn** [comment]
Were you having the issue outside of Unity?

**Comment by reveriejake** [comment]
I haven't tested it outside of Unity, but I can try Unreal either tonight or sometime this weekend to see if it suffers a similar issue.

**Comment by Auburn** [comment]
I'm not too worried about testing other compilers, I think everyone who reported the issue has been using Unity. I just thought from your wording that you weren't using Unity.

According to this: https://docs.unity3d.com/Manual/SL-PragmaDirectives.html

`UNITY_VERSION` is defined

**Comment by reveriejake** [comment]
Hmm, I must have overlooked the UNITY_VERSION define on that page, I did a quick skim. I just tried it out and this one seems to have worked.

**Comment by Auburn** [comment]
Instead of the current defines could you change it to this:
```
#define FNL_FLATTEN [flatten]
#define FNL_UNROLL [unroll(1)]
```

and define them as empty when not on Unity, assuming empty defines are allowed on HLSL?

**Comment by reveriejake** [comment]
Yep I was just working on that. Here is what I put instead.

`#if UNITY_VERSION
    #define FNL_FLATTEN [flatten]
    #define FNL_UNROLL [unroll(1)]
#else
    #define FNL_FLATTEN
    #define FNL_UNROLL
#endif`

Leaving the #else off causes issues in Unity when not included. So not sure if that is an issue, so I included the #else.

**Comment by reveriejake** [comment]
All changes have been pushed

File: HLSL/FastNoiseLite.hlsl [hunk 1]
```diff
@@ -47,7 +47,7 @@
 // VERSION: 1.0.1
 // https://github.com/Auburn/FastNoise
 
-// Switch between using floats or doubles for input position
+// [flatten] switch between using floats or doubles for input position
```
Comment (Auburn) [inline HLSL/FastNoiseLite.hlsl:50]: This looks like an overzealous find and replace

Comment (reveriejake) [reply HLSL/FastNoiseLite.hlsl:50]: You're too good at this.

Comment (reveriejake) [reply HLSL/FastNoiseLite.hlsl:50]: At least it was commented out ;). Let me restore that line.

**Comment by Auburn** [comment]
Thanks!

**Pull request merged**